tools/json-api-client/src/transport/json.ts

```diff
@@ -137,25 +137,27 @@ async function parseResponse<T>(response: Response): Promise<T> {
   } else if (response.status === 401) {
     throw new AuthenticationError(response)
   } else if (response.status >= 500) {
-    const json = await response.json()
-    let errors: ErrorItem[]
-
-    if (json.errors) {
-      errors = json.errors.map((e: ErrorsObject) => ({
-        status: response.status,
-        detail: e.detail,
-      }))
-    } else {
-      errors = [
-        {
-          status: response.status,
-          detail: response.statusText,
-        },
-      ]
-    }
-
+    const errors = await errorItems(response)
     throw new ServerError(errors)
   } else {
     throw new UnknownResponseError(response)
   }
 }
+
+async function errorItems(response: Response): Promise<ErrorItem[]> {
+  const json = await response.json()
```

Comment:
Can we move this call into the caller function and keep this function synchronous? It'll make this function nice and simple

Reply:
Can you expand on how you see that working? It would also require pulling off the attributes for `status` + `statusText` and would result in an extra line each time it's used e.g.

```
// ---
const json = await response.json()
const errors = errorItems(json, response.status, response.statusText)
throw new ServerError(errors)

// ---
const json = await response.json()
const errors = errorItems(json, response.status, response.statusText)
throw new UnprocessableError(errors)
```

vs

```
// ---
const errors = await errorItems(response)
throw new ServerError(errors)

// ---
const errors = await errorItems(response)
throw new UnprocessableError(errors)
```

Reply:
I see, lets just keep the async version then, its much cleaner.